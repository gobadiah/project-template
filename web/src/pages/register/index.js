import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
