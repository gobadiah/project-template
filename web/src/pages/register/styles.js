import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
