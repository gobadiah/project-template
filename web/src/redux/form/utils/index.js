import React from 'react';
import { arrayOf, bool, func, string } from 'prop-types';

import { SubmissionError } from 'redux-form';

import _ from 'lodash';

import { PureComponent } from '~/components/base';

/** Simply translate the error(s) without any formatting. */
export const formatError = (error, t) => (Array.isArray(error) ?
  _.map(error, err => t(err)) :
  t(error));


/** Simply translate the warning(s) without any formatting. */
export const formatWarning = formatError;

/** Handle form errors */
export const handleFormErrors = (err) => {
  if (!err || !err.response || !err.response.status || err.response.status !== 400) {
    throw err;
  }
  const errors = {
    _error: err.response.data.non_field_errors,
    ...err.response.data.errors,
  };
  throw new SubmissionError(errors);
};

/** Base form */
export class BaseForm extends PureComponent {
  getFields = () => {};

  render() {
    const {
      error,
      formatError, // eslint-disable-line no-shadow
      formatWarning, // eslint-disable-line no-shadow
      handleSubmit,
      pristine,
      reset,
      submitting,
      warning,
    } = this.props;
    const { t } = this.context;
    return (
      <form onSubmit={handleSubmit}>
        {this.getFields()}
        <input type='submit' value={t('Send')} disabled={submitting} />
        <input type='button' value={t('Reset')} disabled={pristine || submitting} onClick={reset} />
        {formatWarning(warning, t)}
        {formatError(error, t)}
      </form>
    );
  }
}

BaseForm.propTypes = {
  error: arrayOf(string),
  formatError: func,
  formatWarning: func,
  handleSubmit: func.isRequired,
  pristine: bool.isRequired,
  submitting: bool.isRequired,
  reset: func.isRequired,
  warning: arrayOf(string),
};

BaseForm.defaultProps = {
  error: [],
  formatError,
  formatWarning,
  warning: [],
};
