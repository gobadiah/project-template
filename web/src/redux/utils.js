import { createActions as ca } from 'redux-actions';

export const createActions = (actions, path) => ca({
  web: {
    [path]: actions,
  },
}).web[path];

export const a = 1;
