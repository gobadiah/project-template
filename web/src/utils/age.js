import moment from 'moment';

export default (t, birthday) => t('age', { years: moment().diff(birthday, 'years') });
