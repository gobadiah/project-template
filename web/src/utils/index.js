export { default as unauthorizedHandler } from './unauthorized-handler';
export { default as redirect } from './redirect';
