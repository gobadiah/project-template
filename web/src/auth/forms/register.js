import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
