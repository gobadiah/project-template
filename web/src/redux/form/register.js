import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
