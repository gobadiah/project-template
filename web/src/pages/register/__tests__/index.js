import 'jsdom-global/register';

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { pageProps } from '../../__tests__/utils';

import Register from '..';

describe('Register', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Register {...pageProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shallow rendering should contain RegisterForm', () => {
    const wrapper = shallow(<Register {...pageProps} />);
    wrapper.contains('RegisterForm');
  });
});
