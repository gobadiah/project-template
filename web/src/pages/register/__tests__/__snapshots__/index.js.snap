// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<form
  onSubmit={[Function]}
>
  <div
    className="emotion-0 emotion-1"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      autoComplete="email"
      className={undefined}
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="email"
      value=""
    />
  </div>
  <div
    className="emotion-0 emotion-1"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoComplete="new-password"
      className={undefined}
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="password"
      value=""
    />
  </div>
  <div
    className="emotion-0 emotion-1"
  >
    <label
      htmlFor="confirm_password"
    >
      Password confirmation
    </label>
    <input
      autoComplete="new-password"
      className={undefined}
      name="confirm_password"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="password"
      value=""
    />
  </div>
  <div
    className="emotion-0 emotion-1"
  >
    <label
      htmlFor="first_name"
    >
      First name
    </label>
    <input
      autoComplete="given-name"
      className={undefined}
      name="first_name"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="text"
      value=""
    />
  </div>
  <div
    className="emotion-0 emotion-1"
  >
    <label
      htmlFor="last_name"
    >
      Last name
    </label>
    <input
      autoComplete="family-name"
      className={undefined}
      name="last_name"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="text"
      value=""
    />
  </div>
  <div
    className="emotion-0 emotion-1"
  >
    <label
      htmlFor="birthday"
    >
      Birthday
    </label>
    <input
      autoComplete="bday"
      className={undefined}
      name="birthday"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      type="date"
      value=""
    />
  </div>
  <input
    disabled={false}
    type="submit"
    value="Send"
  />
  <input
    disabled={true}
    onClick={[Function]}
    type="button"
    value="Reset"
  />
</form>
`;
